test/integration_multithread_test.py

```diff
@@ -0,0 +1,82 @@
+from os import system
+from threading import Thread
+import socket
+import time
+import sys
+import subprocess
+
+
+def client_thread_paused(args):
+	TCP_IP = '127.0.0.1'
+	TCP_PORT = 2020
+	BUFFER_SIZE = 1024
+	MESSAGE = "GET /echo HTTP/1.1"
+
+	s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+	s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+	s.connect(('localhost', TCP_PORT))
+	s.send(MESSAGE)
+	time.sleep(5)
```

Comment:
why not just use blocking handler? sleeps are a little less robust in the event that code you expect to run instantly ends up taking longer than the sleep due to some blocking. very unlikely i do admit, but i believe people usually stay away from sleeps for testing